web3/module.py

```diff
@@ -34,12 +44,18 @@ def apply_result_formatters(
         return result
 
 
+TReturn = TypeVar('TReturn')
+
+
 @curry
 def retrieve_blocking_method_call_fn(
-    w3: "Web3", module: Union["Module", "ModuleV2"], method: Method[Callable[..., Any]]
-) -> Callable[..., RPCResponse]:
-    def caller(*args: Any, **kwargs: Any) -> RPCResponse:
-        (method_str, params), response_formatters = method.process_params(module, *args, **kwargs)
+    w3: "Web3", module: Union["Module", "ModuleV2"], method: Method[Callable[..., TReturn]]
+) -> Callable[..., Union[TReturn, LogFilter]]:
+    def caller(*args: Any, **kwargs: Any) -> Union[TReturn, LogFilter]:
+        try:
+            (method_str, params), response_formatters = method.process_params(module, *args, **kwargs)  # noqa: E501
+        except _UseExistingFilter as err:
+            return LogFilter(eth_module=module, filter_id=err.filter_id)
```

Comment:
@pipermerriam I can't figure out how to pull this out without going back to passing around a string method name, since all of the formatters are chosen based on the method name. Let me know if you see another way!

Reply:
If we can't get this out, lets consider making it generic.  I'd suggest doing that as a follow up to this PR.

Roughly speaking, it looks like we need an official way for a `Method` to short-circut the JSON-RPC request and to return a value that is generated locally.  That could be using something like an exception like we did here, but I think that's probably not the best choice here because it feels like a heavy way to implement what is effectively an `if/else`.

We might need to return something more *structured* from `process_params` that we can then switch on.

```python
request = method.preprocess_request(module, *args, **kwargs)
if request.is_json_rpc:
    ...  # do the actual request
elif request.has_static_response:
    ...  # directly return the response value.
else:
    ...  # dunno, maybe just raise Exception("Invariant")
```

I don't know if this is going to work, or be clean.  Just my stab at what the API could look like.  I think you'll be doing yourself and everyone on the team a favor to spend some time working on these APIs, especially in making them less cryptic in how they work.

Reply:
Yeah, agreed. I'll do some mucking around with it in another PR.